tests/Unit/Core/Util/String/StringModifierTest.php

```diff
@@ -118,4 +119,31 @@ public function getNotTooLongStringsForEndCutting(): Generator
         yield ['testable', 20];
         yield ['good bye cruel world 10.99', 128];
     }
+
+    /**
+     * @dataProvider getDataForCamelCaseToKebabCaseConversion
+     *
+     * @param string $input
+     * @param string $expectedOutput
+     */
+    public function testItConvertsCamelCaseToKebabCase(string $input, string $expectedOutput): void
+    {
+        $output = $this->stringModifier->convertCamelCaseToKebabCase($input);
+
+        Assert::assertSame($expectedOutput, $output);
+    }
+
+    /**
+     * @return Generator
+     */
+    public function getDataForCamelCaseToKebabCaseConversion(): Generator
+    {
+        yield ['HelloWorld', 'hello-world'];
+        yield ['helloWorld', 'hello-world'];
+        yield ['goodbyecruelWorld', 'goodbyecruel-world'];
+        yield ['goodbyecruelWORLD', 'goodbyecruel-world'];
+        yield ['GoodBYEcruelWORLD', 'good-byecruel-world'];
```

Comment:
I'm not sure because I'm not familiar with kebab case but in this case isn't the expected result `yield ['GoodBYEcruelWORLD', 'good-bye-cruel-world'];`

Since we go from upper to lower case?

Reply:
Im just not willing to spend more time for such cases as I doubt it will ever occur (btw `GoodBYEcruelWORLD` doesn't seem to fit into any standards, so there cannot be a standard how to convert it to another standard :sweat_smile: ) , so I just added this case as "how it works now" :smile:

Reply:
Yeah you're right! if we have problems we can always improve it later anyway